server/remote_asset/fetch_server/fetch_server.go

```diff
@@ -116,12 +114,12 @@ func (p *FetchServer) FetchBlob(ctx context.Context, req *rapb.FetchBlobRequest)
 	for _, qualifier := range req.GetQualifiers() {
 		var prefix string
 		if qualifier.GetName() == ChecksumQualifier {
-			if strings.HasPrefix(qualifier.GetValue(), sha256Prefix) {
-				checksumFunc = repb.DigestFunction_SHA256
-				prefix = sha256Prefix
-			} else if strings.HasPrefix(qualifier.GetValue(), blake3Prefix) {
-				checksumFunc = repb.DigestFunction_BLAKE3
-				prefix = blake3Prefix
+			for _, digestFunc := range []string{"sha1", "sha256", "sha384", "sha512", "blake3"} {
+				if strings.HasPrefix(qualifier.GetValue(), digestFunc+"-") {
```

Comment:
Move `digestFunc+"-"` to a variable outside the `if` statement so you would only compute it once.